lib/pages/chat/empty_chat.dart

```diff
@@ -0,0 +1,396 @@
+                return const Center(
+                  child: CircularProgressIndicator(),
+                );
+              }
+            );
+          }
+        });
+    });
+  }
+
+  void _openDownloadedFileForPreview({
+    required DownloadFileForPreviewResponse downloadFileForPreviewResponse
+  }) async {
+    final mimeType = downloadFileForPreviewResponse.mimeType;
+    final openResults = await OpenFile.open(
+      downloadFileForPreviewResponse.filePath,
+      type: mimeType,
+      uti: DocumentUti(SupportedPreviewFileTypes.iOSSupportedTypes[mimeType]).value
+    );
+    Logs().d('ChatController:_openDownloadedFileForPreview(): ${openResults.message}');
+
+    if (openResults.type != ResultType.done) {
+      await Share.shareXFiles([XFile(downloadFileForPreviewResponse.filePath)]);
+      return;
+    }
+  }
+
+  void openVideoCameraAction() async {
+
+  }
+
+  void sendStickerAction() async {
+    
+  }
+
+  void voiceMessageAction() async {
+    
+  }
+
+  void emojiPickerAction() {
+    if (showEmojiPicker) {
+      inputFocus.requestFocus();
+    } else {
+      inputFocus.unfocus();
+    }
+    emojiPickerType = EmojiPickerType.keyboard;
+    setState(() => showEmojiPicker = !showEmojiPicker);
+  }
+
+  void _inputFocusListener() {
+    if (showEmojiPicker && inputFocus.hasFocus) {
+      emojiPickerType = EmojiPickerType.keyboard;
+      if (mounted) {
+        setState(() => showEmojiPicker = false);
+      }
+    }
+  }
+
+  void sendLocationAction() async {
+    
+  }
+
+  void scrollDown() {
+    if (scrollController.hasClients) {
+      scrollController.jumpTo(0);
+    }
+  }
+
+  void onEmojiBottomSheetSelected(_, Emoji? emoji) {
+    typeEmoji(emoji);
+    onInputBarChanged(sendController.text);
+  }
+
+  void typeEmoji(Emoji? emoji) {
+    if (emoji == null) return;
+    final text = sendController.text;
+    final selection = sendController.selection;
+    final newText = sendController.text.isEmpty
+        ? emoji.emoji
+        : text.replaceRange(selection.start, selection.end, emoji.emoji);
+    sendController.value = TextEditingValue(
+      text: newText,
+      selection: TextSelection.collapsed(
+        // don't forget an UTF-8 combined emoji might have a length > 1
+        offset: selection.baseOffset + emoji.emoji.length,
+      ),
+    );
+  }
+
+  late Iterable<Event> _allReactionEvents;
+
+  void emojiPickerBackspace() {
+    sendController
+      ..text = sendController.text.characters.skipLast(1).toString()
+      ..selection = TextSelection.fromPosition(
+        TextPosition(offset: sendController.text.length),
+      );
+  }
+
+  void pickEmojiReactionAction(Iterable<Event> allReactionEvents) async {
+    _allReactionEvents = allReactionEvents;
+    emojiPickerType = EmojiPickerType.reaction;
+    setState(() => showEmojiPicker = true);
+  }
+
+  void onInputBarSubmitted(_) {
+    send();
+    FocusScope.of(context).requestFocus(inputFocus);
+  }
```

Comment:
```suggestion
  void onInputBarSubmitted() {
    send();
    FocusScope.of(context).requestFocus(inputFocus);
  }
```

Reply:
onSubmitted need function have parameter, don't worry, when we need to use para, just remove _